test/slex_lexer_test.exs

```diff
@@ -0,0 +1,81 @@
+defmodule SlexLexerTest do
+  use ExUnit.Case
+
+  def assert_tokens(input, tokens) do
+    {:ok, output, _} = :slex_lexer.string(input)
+    assert output == tokens
+  end
+
+  test "single element" do
+    assert_tokens 'html', [
+      { :word,      1, 'html' }
+    ]
+  end
+
+  test "single element with id" do
+    assert_tokens 'div#id', [
+      { :word,      1, 'div'  },
+      { :id,        1, 'id'   }
+    ]
+  end
+
+  test "single element with class" do
+    assert_tokens 'div.class', [
+      { :word,      1, 'div'   },
+      { :class,     1, 'class' }
+    ]
+  end
+
+  test "single element with multiple classes" do
+    assert_tokens 'div.c1.c-2', [
+      { :word,      1, 'div'   },
+      { :class,     1, 'c1'    },
+      { :class,     1, 'c-2'   }
+    ]
+  end
+
+  test "element with one attribute" do
+    assert_tokens 'meta a="A"', [
+      { :word,      1, 'meta' },
+      { :space,     1, ' '    },
```

Comment:
Agreed ideally spaces should be skipped, but I had issues making anchoring indents to the start of the line. Mainly because leex doesn't support the start of line anchoring... Might be another way to do it, or we just write our own lexer to produce the token stream we want.